src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -520,47 +517,40 @@ private bool CheckCopiedOutputFiles(Log log, in TimestampCache timestampCache, S
             return true;
         }
 
-        private bool CheckCopyToOutputDirectoryFiles(Log log, in TimestampCache timestampCache, State state)
+        private bool CheckPreserveNewestCopyToOutputDirectoryFiles(Log log, in TimestampCache timestampCache, State state)
         {
-            IEnumerable<(string path, string? link, CopyType copyType)> items = state.ItemsByItemType.SelectMany(kvp => kvp.Value).Where(item => item.copyType == CopyType.CopyIfNewer);
+            IEnumerable<KeyValuePair<string, (string targetPath, CopyType copyType)>> preserveNewestItems
+                = state.CopyToOutputDirectoryItems.Where(item => item.Value.copyType == CopyType.PreserveNewest);
 
             string outputFullPath = Path.Combine(state.MSBuildProjectDirectory, state.OutputRelativeOrFullPath);
 
-            foreach ((string path, string? link, _) in items)
+            foreach ((string sourcePath, (string targetFilename, _)) in preserveNewestItems)
```

Comment:
Nit: We often use "fullPath" to indicate this a rooted path, "sourceFullPath".

Reply:
I will change the variable name to sourceFullPath to make it clear.